packages/clay-slider/src/index.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React, {useEffect, useRef, useState} from 'react';
+
+interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
+	/**
+	 * Flag that will disable component features.
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Sets maximum permitted value.
+	 */
+	max?: number;
+
+	/**
+	 * Sets minimum permitted value
+	 */
+	min?: number;
+
+	/**
+	 * Callback will always be called when the slider value changes.
+	 */
+	onValueChange: (value: number) => void;
+
+	/**
+	 * Flag to show tooltip or not.
+	 */
+	showTooltip?: boolean;
+
+	/**
+	 * Sets stepping interval.
+	 */
+	step?: number;
+
+	/**
+	 * Set tooltip position.
+	 */
+	tooltipPosition?: 'top' | 'bottom';
+
+	/**
+	 * Set the current value of the slider.
+	 */
+	value: number;
+}
+
+const THUMB_WIDTH = 24;
```

Comment:
@matuzalemsteles This should be configurable. I was hoping to be able to grab the thumb width via js, but because it's in the shadow dom I wasn't able to quickly. Also the thumb selector is different for every browser which adds to the complexity.

Looking back now, I can probably add a thumb overlay (similar to the tooltip) to make this easier and forget about trying to style the browser default thumb. I can send an update to the slider markup. It should make it more customizable.

Reply:
Oh thanks @pat270, for the info, I think adding a thumb overlay is a better solution. Even if we make it configurable people will probably have to have a list of widths that the thumb can be, I think it's not good to add this complexity to the component.

Reply:
hey guys, just keep in mind that the less we use of the native input the more we need to build back in. Native thumb will have builtin accessibility, keyboard navigation and so on... the more we can keep the better

Reply:
@jbalsas it will still be using the native input and should work the same. The only thing we need to calculate correctly is the progress bar width.

Reply:
hey @pat270 with your changes will i need to change anything here? Let me know if I need to wait for your changes to come in first.

Reply:
@matuzalemsteles yes, I'm building on top of this pr. The only thing that needs to change is `THUMB_WIDTH` should get the width of `.clay-range-thumb` and the tooltip needs to be nested inside `.clay-range-thumb`. I'll send my changes to you when I finish.

Reply:
@pat270 ok, I just made the small updates.

Reply:
Awesome, @pat270! I thought we would need to control the thumb, but I see now that it's just a placeholder to figure out the thumb's native width?